### PR TITLE
Add tan_cuda for complex dtypes

### PR DESCRIPTION
**Pull request by anjali411**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#38400 Add tan_cuda for complex dtypes**

Differential Revision: [D21572209](https://our.internmc.facebook.com/intern/diff/D21572209)

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -17713,6 +17715,7 @@ def inner(self, device, dtype):
     ('rsqrt', '', lambda t, d: _small_3d(t, d) + 1, lambda t, d: [], 1e-2, 1e-5, 1e-4, _float_types_no_half),
     ('sinh', '', lambda t, d: _small_3d(t, d).clamp(-1, 1), lambda t, d: [], 1e-3, 1e-5, 1e-5, _float_types),
     ('tan', '', lambda t, d: _small_3d(t, d).clamp(-1, 1), lambda t, d: [], 1e-3, 1e-5, 1e-5, _float_types),
+    ('tan', 'complex', lambda t, d: _small_3d(t, d), lambda t, d: [], 1e-3, 1e-5, 1e-5, _complex_types)
```
Comment (albanD) [inline test/test_torch.py:17718]: I am surprised that `_complex_types` did not exist yet here. How do we test the other methods that already support complex?

File: test/test_autograd.py [hunk 2]
```diff
@@ -4185,7 +4185,7 @@ def run_functional_checks(test_case, test_name, name, apply_fn, run_grad_checks,
         test_case.assertEqualTypeString(self_variable, self_variable.grad)
         test_case.assertEqual(self_variable.size(), self_variable.grad.size())
 
-separate_complex_tests = ['log', 'log10', 'log1p', 'log2', 'reciprocal']
+separate_complex_tests = ['log', 'log10', 'log1p', 'log2', 'reciprocal', 'tan']
```
Comment (albanD) [inline test/test_autograd.py:4188]: Why is this one in the reparate list? It should work for real numbers as well no?

Comment (anjali411) [reply test/test_autograd.py:4188]: the test for floating points uses clamp in the input and clamp does not work for complex

File: test/test_torch.py (hunk 1 above)
Comment (anjali411) [reply test/test_torch.py:17718]: Unfortunately there are no existing tests for these ops for complex dtypes. we should add it soon though ...

**Review by albanD — APPROVED** [review]
LGTM

**Pull request closed without merging**